ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -6,122 +6,152 @@
 module Ouroboros.Network.Diffusion
   ( -- * Common API
     P2P (..)
-  , DiffusionTracer (..)
-  , Tracers (..)
-  , nullTracers
   , ExtraTracers (..)
-  , Failure (..)
-  , Arguments (..)
-  , ExtraArguments (..)
+  , ArgumentsExtra (..)
   , Applications (..)
-  , ExtraApplications (..)
+  , ApplicationsExtra (..)
     -- * Run data diffusion
   , run
-    -- * Re-exports
-  , P2P.AbstractTransitionTrace
-  , PublicPeerSelectionState
-  , makePublicPeerSelectionStateVar
   ) where
 
-import Control.Exception (IOException)
-import Data.Functor (void)
-
-import Network.Socket (Socket)
-
-import Ouroboros.Network.NodeToClient (LocalAddress, LocalSocket,
-           NodeToClientVersion, NodeToClientVersionData)
-import Ouroboros.Network.NodeToNode (NodeToNodeVersion, NodeToNodeVersionData,
-           RemoteAddress)
-import Ouroboros.Network.PeerSelection.Governor.Types
-
+import Cardano.Diffusion.P2P qualified as P2P
 import Cardano.Node.ArgumentsExtra (CardanoArgumentsExtra)
 import Cardano.Node.LedgerPeerConsensusInterface
            (CardanoLedgerPeersConsensusInterface)
 import Cardano.Node.PeerSelection.Governor.PeerSelectionState
            (CardanoPeerSelectionState)
 import Cardano.Node.PeerSelection.PeerTrustable (PeerTrustable)
 import Cardano.Node.PublicRootPeers (CardanoPublicRootPeers)
-import Ouroboros.Network.Diffusion.Common as Common
+import Control.Exception (IOException)
+import Data.Functor (void)
+import Network.Socket (Socket)
+import Ouroboros.Network.Diffusion.Common (Arguments, Tracers)
+import Ouroboros.Network.Diffusion.Common qualified as Common
 import Ouroboros.Network.Diffusion.NonP2P qualified as NonP2P
-import Ouroboros.Network.Diffusion.P2P qualified as P2P
-import Ouroboros.Network.PeerSelection.LedgerPeers.Type (mapExtraAPI)
+import Ouroboros.Network.NodeToClient (LocalAddress, LocalSocket,
+           NodeToClientVersion, NodeToClientVersionData)
+import Ouroboros.Network.NodeToNode (NodeToNodeVersion, NodeToNodeVersionData,
+           RemoteAddress)
 
 -- | Promoted data types.
 --
-data P2P = P2P | NonP2P
+data P2P = P2P        -- ^ General P2P mode. Can be instantiated with custom
+                      -- data types
+         | NonP2P     -- ^ Cardano non-P2P mode. Deprecated
+         | P2PCardano -- ^ Cardano P2P mode.
```

Comment:
Do we really need `P2PCardano` different from `P2P`? Or does this disappear by the end of the commit list (I go patch by patch).

ref: a382ed87d2998ff90dd825ddbc6ac2ccbdac8da0

Reply:
This disappears.